target/riscv/cpu_helper.c

```diff
@@ -681,8 +696,24 @@ static int get_physical_address(CPURISCVState *env, hwaddr *physical,
                 *prot |= PAGE_WRITE;
             }
 #if defined(TARGET_CHERI) && !defined(TARGET_RISCV32)
-            if ((pte & PTE_LC) == 0)
-                *prot |= PAGE_LC_CLEAR;
+            if ((pte & PTE_CR) == 0) {
+                if ((pte & PTE_CRM) == 0) {
+                    *prot |= PAGE_LC_CLEAR;
+                } else {
+                    *prot |= PAGE_LC_TRAP;
+                }
+            } else {
+                if (pte & PTE_CRM) {
+                    // Cap-loads checked against GCLG in CCSR using PTE_U
```

Comment:
```suggestion
                    /* Cap-loads checked against GCLG in CCSR using PTE_U */
```
C++-style comments plague the CHERI parts but upstream style is to use C-style ones.